dynamodb-enhanced/runtime/src/main/java/io/quarkus/amazon/dynamodb/enhanced/runtime/DynamodbEnhancedClientProducer.java

```diff
@@ -0,0 +1,50 @@
+package io.quarkus.amazon.dynamodb.enhanced.runtime;
+
+import javax.annotation.PreDestroy;
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Produces;
+
+import io.quarkus.arc.Arc;
+import software.amazon.awssdk.enhanced.dynamodb.DynamoDbEnhancedAsyncClient;
+import software.amazon.awssdk.enhanced.dynamodb.DynamoDbEnhancedClient;
+import software.amazon.awssdk.services.dynamodb.DynamoDbAsyncClient;
+import software.amazon.awssdk.services.dynamodb.DynamoDbClient;
+
+@ApplicationScoped
+public class DynamodbEnhancedClientProducer {
+
+    private final DynamoDbEnhancedClient syncEnhancedClient;
+
+    private final DynamoDbEnhancedAsyncClient asyncEnhancedClient;
+
+    DynamodbEnhancedClientProducer() {
+        this.syncEnhancedClient = DynamoDbEnhancedClient.builder()
+                .dynamoDbClient(Arc.container().instance(DynamoDbClient.class).get()).build();
+        this.asyncEnhancedClient = DynamoDbEnhancedAsyncClient.builder()
+                .dynamoDbClient(Arc.container().instance(DynamoDbAsyncClient.class).get()).build();
+    }
+
+    @Produces
+    @ApplicationScoped
+    public DynamoDbEnhancedClient client() {
+        if (syncEnhancedClient == null) {
+            throw new IllegalStateException("The DynamoDbEnhancedClient is required but has not been detected/configured.");
+        }
+        return syncEnhancedClient;
+    }
+
+    @Produces
+    @ApplicationScoped
+    public DynamoDbEnhancedAsyncClient asyncClient() {
+        if (asyncEnhancedClient == null) {
+            throw new IllegalStateException(
+                    "The DynamoDbEnhancedAsyncClient is required but has not been detected/configured.");
+        }
+        return asyncEnhancedClient;
+    }
+
+    @PreDestroy
+    public void destroy() {
+
+    }
```

Comment:
I think get on `Arc.container().instance()` will throw if not registered. BTW, you may get rid of Arc with `Instance` or by injecting the client in the producer. I would favor the latest as the extension do not need to keep a direct reference on the enhanced client instances.

```suggestion
    private final DynamoDbEnhancedClient syncEnhancedClient;

    DynamodbEnhancedClientProducer(Instance<DynamoDbClient> syncClientInstance) {
        this.syncEnhancedClient = syncClientInstance.isResolvable() ? DynamoDbEnhancedClient.builder()
        .dynamoDbClient(syncClientInstance.get()).build() : null;
    }

    @Produces
    @ApplicationScoped
    public DynamoDbEnhancedClient client() {
        if (syncEnhancedClient == null) {
            throw new IllegalStateException("The DynamoDbEnhancedClient is required but has not been detected/configured.");
        }
        return syncEnhancedClient;
    }

    @Produces
    @ApplicationScoped
    public DynamoDbEnhancedAsyncClient asyncClient(DynamoDbAsyncClient client) {
        return DynamoDbEnhancedAsyncClient.builder()
                .dynamoDbClient(client).build();
    }
```

Reply:
Thanks, i can switch to Instance<...> but I can not Inject DynamoDbClient and DynamoDbAsyncClient. If I do that I always get `Unable to load region from any of the providers in the chain ....`

Reply:
Any idea or is Instance<...> okay?